frontend/pkg/database/cache.go

```diff
@@ -0,0 +1,52 @@
+package database
+
+import "context"
+
+var _ DBClient = &Cache{}
+
+type Cache struct {
+	cluster      map[string]*HCPOpenShiftClusterDocument
+	subscription map[string]*SubscriptionDocument
+}
+
+func NewCache() *Cache {
```

Comment:
Suggest returning `DBClient` here, since it doesn't look like you can do anything with a `Cache` beyond what the `DBClient` interface provides.